examples/ConfigurationStoreBackup/BackupAppConfigurationStore.cs

```diff
@@ -0,0 +1,166 @@
+        {
+            HashSet<KeyLabel> updatedKeyLabels = new HashSet<KeyLabel>();
+            foreach (QueueMessage message in retrievedMessages)
+            {
+                string decodedMessage = string.Empty;
+                try
+                {
+                    // Event grid will encode events in Base64 format before publishing to storage queue. Decode the message before parsing it.
+                    decodedMessage = Encoding.UTF8.GetString(Convert.FromBase64String(message.MessageText));
+                    JToken dataObject = JObject.Parse(decodedMessage).Property("data")?.Value;
+                    if (dataObject != null && dataObject["key"] != null)
+                    {
+                        string key = dataObject["key"].ToString();
+                        string label = dataObject["label"]?.ToString();
+                        updatedKeyLabels.Add(new KeyLabel(key, label));
+                    }
+                }
+                catch (FormatException)
+                {
+                    // If its not in valid base64 format, ignore the queue message. 
+                    log.LogInformation($"Queue message in invalid Base64 format will be ignored.\nMessage: {message.MessageText}");
+                }
+                catch (JsonReaderException)
+                {
+                    // If its not a valid JSON, ignore the queue message. 
+                    log.LogInformation($"Queue message in invalid JSON format will be ignored.\nMessage: {decodedMessage}");
+                }
+            }
+            return updatedKeyLabels;
+        }
+
+        private static async Task<bool> BackupKeyValuesAsync(HashSet<KeyLabel> updatedKeyLabels, ILogger log)
+        {
+            bool backupCompleted = false;
+            try
+            {
+                // Read all settings from primary store and update secondary store.
+                await foreach (ConfigurationSetting setting in primaryAppConfigClient.GetConfigurationSettingsAsync(new SettingSelector()).ConfigureAwait(false))
+                {
+                    KeyLabel primaryStoreKeyLabel = new KeyLabel(setting.Key, setting.Label);
+                    if (updatedKeyLabels.Contains(primaryStoreKeyLabel))
+                    {
+                        // Current setting retrieved from primary store needs to be updated in secondary store.
+                        await secondaryAppConfigClient.SetConfigurationSettingAsync(setting).ConfigureAwait(false);
+                        log.LogInformation($"Successfully updated key: {setting.Key} label: {setting.Label}");
+                        
+                        updatedKeyLabels.Remove(primaryStoreKeyLabel);
+                        if (updatedKeyLabels.Count == 0)
+                        {
+                            // If no more key labels were updated, we can stop processing.
```

Comment:
> If no more key labels were updated, we can stop processing. [](start = 30, length = 60)

This condition is already true inside the if block. Not sure if this comment is needed.